### PR TITLE
Technical limitations of how old one can be on D* #5373

### PR DESCRIPTION
**Pull request by aka001**
Please review the pull request.

### CODE REVIEW

File: app/views/profiles/_edit.haml [hunk 1]
```diff
@@ -35,7 +36,7 @@
     = t('profiles.edit.your_birthday')
 
   .row-fluid
-    = select_date profile.birthday, { :prompt => true, :default => true, :order => t('date.order'), :start_year => 2000, :end_year => 1910, :prefix => 'profile[date]' }, { :class => 'span4' }
+    = select_date profile.birthday, { :prompt => true, :default => true, :order => t('date.order'), :start_year => val, :end_year => 1910, :prefix => 'profile[date]' }, { :class => 'span4' }
```
Comment (jhass) [inline app/views/profiles/_edit.haml]: How about putting just `13.years.ago.year` here directly? See http://api.rubyonrails.org/classes/ActiveSupport/Duration.html#method-i-ago

**Comment by jhass** [comment]
Actually it would be nice to read the minimum age from the config, like it's done in terms page and set the maximum age to 125 years, like I mentioned in #5373

File: app/views/profiles/_edit.haml (hunk 1 above)
Comment (aka001) [inline app/views/profiles/_edit.haml]: @jhass Thanks for a quick reply. I have updated the pr. The line ```13.years.ago``` works.

**Comment by aka001** [comment]
@jhass Okay. I am working on it.

**Comment by aka001** [comment]
@jhass I have updated the pr. Kindly look into it. Thank you for all your help.

File: app/views/profiles/_edit.haml [hunk 2]
```diff
@@ -2,6 +2,8 @@
 -#   licensed under the Affero General Public License version 3 or later.  See
 -#   the COPYRIGHT file.
 
+- lower_limit = upper_limit_date_of_birth
+- upper_limit = lower_limit_date_of_birth
```
Comment (jhass) [inline app/views/profiles/_edit.haml]: There's no need for these assignments, just call the helpers directly below.

File: app/helpers/profile_helper.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#   Copyright (c) 2010-2011, Diaspora Inc.  This file is
+#   licensed under the Affero General Public License version 3 or later.  See
+#   the COPYRIGHT file.
+
+module ProfileHelper
+  def upper_limit_date_of_birth()
+     minimum_year=AppConfig.age.minimum
```
Comment (jhass) [inline app/helpers/profile_helper.rb]: The key is `AppConfig.settings.terms.minium_age` Also please use spaces around the `=` in assignments.

File: app/helpers/profile_helper.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+     if AppConfig.age.minimum==nil
```
Comment (jhass) [inline app/helpers/profile_helper.rb]: With the correct key, you'll see the default value is `false`, not `nil`. Also reuse the variable from above.

File: app/helpers/profile_helper.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+       minimum_year=13
+     end
+     minimum_year.years.ago.year
+  end
+
+  def lower_limit_date_of_birth()
+     maximum_year=AppConfig.age.maximum
+     if AppConfig.age.maximum==nil
```
Comment (jhass) [inline app/helpers/profile_helper.rb]: I don't think this needs to be configurable, just hardcode `125.years.ago.year`

**Comment by aka001** [comment]
@jhass  I have updated the pull request according to your comments. Kindly review it.

**Comment by aka001** [comment]
@houndci I have updated the pull request and squashed all of them into one. I hope it's fine now.
Thank you.

File: app/helpers/profile_helper.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+#   the COPYRIGHT file.
+
+module ProfileHelper
+  def upper_limit_date_of_birth
+    minimum_year = AppConfig.settings.terms.minimum_age.get
+    if !(minimum_year)
+      minimum_year = 13
+    else
+      minimum_year = minimum_year.to_i
+    end
```
Comment (jhass) [inline app/helpers/profile_helper.rb]: Okay, just one last thing:

Let's prefer `unless` over `if !`:

```ruby
unless minimum_year
  minimum_year = 13
else
  minimum_year = minimum_year.to_i
end
```
Let's prefer `if` over `unless` when we can swap branches:


```ruby
if minimum_year
  minimum_year = minimum_year.to_i
else
  minimum_year = 13
end
```

Let's extract the assignment:

```ruby
 minimum_year = if minimum_year
  minimum_year.to_i
else
  13
end
```

And reduce it to a ternary:
```ruby
 minimum_year = minimum_year ?  minimum_year.to_i : 13
```

Comment (aka001) [inline app/helpers/profile_helper.rb]: Thank you for the explanation. I wasn't aware of the convention followed in diaspora.
I have updated the pull request and squashed the commits.

**Comment by jhass** [comment]
Thanks!

**Comment by RoseTheFlower** [comment]
Definitely looks better than the way it is now. Thank you.

**Comment by aka001** [comment]
:)

**Pull request merged**